storage/feed.go

```diff
@@ -177,6 +177,45 @@ func (sef StoreEventFeed) endScanRanges() {
 	sef.f.Publish(&EndScanRangesEvent{sef.id})
 }
 
+// StoreEventListener is an interface that can be implemented by objects which
+// listen for events published by stores.
+type StoreEventListener interface {
+	OnAddRange(event *AddRangeEvent)
+	OnUpdateRange(event *UpdateRangeEvent)
+	OnRemoveRange(event *RemoveRangeEvent)
+	OnSplitRange(event *SplitRangeEvent)
+	OnMergeRange(event *MergeRangeEvent)
+	OnStartStore(event *StartStoreEvent)
+	OnBeginScanRanges(event *BeginScanRangesEvent)
+	OnEndScanRanges(event *EndScanRangesEvent)
+}
+
+// ProcessStoreEvents reads store events from the supplied channel and passes
+// them to the correct methods of the supplied StoreEventListener. This method
+// will run until the Subscription's events channel is closed.
+func ProcessStoreEvents(l StoreEventListener, sub *util.Subscription) {
```

Comment:
ah, if this is what the eventual usage will look like then I take back what I said about concurrency.